src/Model/File.php

```diff
@@ -122,6 +167,125 @@ public function createFromPath(string $path, string $fileName = null): Model
+    {
+        $this->assertIsEntity();
+
+        $max_width = $this->thumbnailMaxWidth;
+        $max_height = $this->thumbnailMaxHeight;
+
+        [$width, $height, $image_type] = getimagesize($path);
+
+        switch ($image_type) {
+            case \IMAGETYPE_GIF:
+                $src = imagecreatefromgif($path);
+
+                break;
+            case \IMAGETYPE_JPEG:
+                $src = imagecreatefromjpeg($path);
+
+                break;
+            case \IMAGETYPE_PNG:
+                $src = imagecreatefrompng($path);
+
+                break;
+            default:
+                return false; // unsupported image type
+        }
+
+        $x_ratio = $max_width / $width;
+        $y_ratio = $max_height / $height;
+
+        if (($width <= $max_width) && ($height <= $max_height)) {
+            $tn_width = $width;
+            $tn_height = $height;
+        } elseif (($x_ratio * $height) < $max_height) {
+            $tn_height = (int) ceil($x_ratio * $height);
+            $tn_width = $max_width;
+        } else {
+            $tn_width = (int) ceil($y_ratio * $width);
+            $tn_height = $max_height;
+        }
+
+        $tmp = imagecreatetruecolor($tn_width, $tn_height);
+
+        // Check if this image is PNG or GIF, then set if Transparent
+        if ($image_type === \IMAGETYPE_PNG || $image_type === \IMAGETYPE_GIF) {
+            imagealphablending($tmp, false);
+            imagesavealpha($tmp, true);
+            $transparent = imagecolorallocatealpha($tmp, 255, 255, 255, 127);
+            imagefilledrectangle($tmp, 0, 0, $tn_width, $tn_height, $transparent);
+        }
+        imagecopyresampled($tmp, $src, 0, 0, 0, 0, $tn_width, $tn_height, $width, $height);
+
+        // create temporary thumb file
+        $thumbFile = tmpfile();
+        switch ($image_type) {
+            case \IMAGETYPE_GIF:
+                imagegif($tmp, $thumbFile);
+
+                break;
+            case \IMAGETYPE_JPEG:
+                imagejpeg($tmp, $thumbFile, 100); // best quality
+
+                break;
+            case \IMAGETYPE_PNG:
+                imagepng($tmp, $thumbFile, 0); // no compression
+
+                break;
+        }
+        $uri = stream_get_meta_data($thumbFile)['uri'];
+
+        // free up memory
+        imagedestroy($src);
+        imagedestroy($tmp);
+
+        // Import it in filestore and link to this (original image) entity
+        $ext = $this->get('meta_extension') ? '.' . $this->get('meta_extension') : '';
+        $thumbName = basename($this->get('meta_filename'), $ext) . '.thumb' . $ext;
+        $thumbModel = $this->getModel();
+        $thumbModel->createThumbnail = false; // do not create thumbnails of thumbnails
+        $thumbEntity = $thumbModel->createFromPath($uri, $thumbName);
+        $thumbEntity->save(['source_file_id' => $this->getId()]);
+
+        // close tmp file and it will be deleted
+        fclose($thumbFile);
```

Comment:
as long as you want to close it, use try/finally

Reply:
I don't see the use of `finally` in php as such, but if you wish so - ok :)